app/models/journeys/navigator.rb

```diff
@@ -0,0 +1,210 @@
+        form_class = journey.form_class_for_slug(slug:)
+        form = form_class.new(
+          journey:,
+          journey_session:,
+          params:,
+          session:
+        )
+
+        if current_slug == slug
+          return last_valid_slug
+        elsif form.respond_to?(:completed?)
+          if form.completed?
+            last_valid_slug = slug
+          end
+        elsif form.valid?
+          last_valid_slug = slug
+        else
+          return last_valid_slug
+        end
+      end
+    end
+
+    # given the current slug
+    # returns the next slug in the sequence
+    # used when a user hits continue on a form
+    def next_slug
+      return "ineligible" if eligibility_checker.ineligible?
+
+      if current_slug.nil?
+        forms.first
+      else
+        forms.each_with_index do |form, index|
+          slug = journey.slug_for_form(form:)
+
+          if form.respond_to?(:completed?)
+            if !form.completed?
+              return slug
+            end
+          elsif form.invalid?
+            return slug
+          end
+
+          if current_slug == slug
+            current_index = forms.index(form)
+            next_index = current_index + 1
+            next_form = forms[next_index]
+            next_slug = journey.slug_for_form(form: next_form)
+
+            return next_slug
+          end
+        end
+      end
+    end
+
+    # is a user allowed to visit the current_slug?
+    # use this to guard against a user jumping ahead in a journey
+    def permissible_slug?
+      if current_slug == "ineligible" && eligibility_checker.ineligible?
+        return true
+      end
+
+      forms.each do |form|
+        slug = journey.slug_for_form(form:)
+
+        if current_slug == slug
+          return true
+        end
+
+        if form.respond_to?(:completed?)
+          if !form.completed?
+            return false
+          end
+        elsif form.invalid?
+          return false
+        end
+      end
+    end
+
+    # ignoring the current_slug
+    # return the last slug the user is allowed to visit
+    # this is the furthest slug available in the journey given their state
+    # use this when user requests a non permissible slug
+    def furthest_permissible_slug
+      last_slug = nil
+
+      forms.each_with_index do |form, index|
+        slug = journey.slug_for_form(form:)
+
+        if form.respond_to?(:completed?)
+          if form.completed?
+            last_slug = slug
+          else
+            return slug
+          end
+        elsif form.valid?
+          last_slug = slug
+        else
+          return slug
+        end
+      end
+
+      last_slug
+    end
+
+    # when a user changes an answer
+    # an existing answer might no longer be applicable
+    # if so we clear all those inaccessbile answers
+    def clear_impermissible_answers
+      ApplicationRecord.transaction do
+        impermissible_forms.each do |form|
+          form.clear_answers_from_session
```

Comment:
Not a blocker, and probably something to implement after we've migrated the other journeys to use the navigator, but I think this class should handle clearing the answers rather than the forms, I don't think forms need to know about clearing answers as it seems like a journey navigation concern. Could probably do something like
```
impermissible_forms.flat_map(&:attribute_names).each do |attribute|
  session.clear_answer!(attribute)
end
```

obv not important!